source/adapters/cuda/enqueue.cpp

```diff
@@ -284,9 +284,15 @@ setKernelParams(const ur_context_handle_t Context,
                                    CudaImplicitOffset);
     }
 
-    if (Context->getDevice()->maxLocalMemSizeChosen()) {
+    auto Device = Context->getDevice();
+    if (LocalSize > static_cast<uint32_t>(Device->getMaxCapacityLocalMem())) {
```

Comment:
do we need this `static_cast`?

Reply:
Without the `static_cast`, we would have here implicit conversion (`Device->getMaxCapacityLocalMem()` is of type `int`) which is also okay...but I think this makes it clearer and more readable what we're comparing. 

Also, using `static_cast` like this is uniform across several parts of this code so I aimed to match with that.